python/ray/dag/dag_node.py

```diff
@@ -335,6 +369,39 @@ def __call__(self, node: "DAGNode"):
             )
         )
 
+    def bfs(self, fn: "Callable[[DAGNode], T]"):
```

Comment:
Not sure if `bfs` is still accurate if the algorithm goes to upstream nodes as well. Maybe just call it `traverse` ?

Reply:
Update to `traverse_and_apply`: https://github.com/ray-project/ray/pull/47261/commits/cfeb6020eb5951916ce4ee9a8db9a20a1cdd04ae